src/components/Navbar/index.tsx

```diff
@@ -1,19 +1,44 @@
 import * as React from 'react';
 import { Button, Navbar } from 'react-bootstrap';
+import { connect } from 'react-redux';
 
 import { gettext } from '../../utils';
-import { Profile } from '../App';
 import LoginButton from '../LoginButton';
+import { logOutFromServer } from '../../api';
+import { ApplicationState, ConnectedReduxProps } from '../../configureStore';
+import { ApiState } from '../../reducers/api';
+import {
+  User,
+  actions as userActions,
+  getCurrentUser,
+} from '../../reducers/users';
 import styles from './styles.module.scss';
 
-type Props = {
-  onLogOut: () => void;
-  profile: null | Profile;
+type PublicProps = {
+  _logOutFromServer: typeof logOutFromServer;
 };
 
+type PropsFromState = {
+  apiState: ApiState;
+  profile: User | null;
+};
+
+type Props = PublicProps & PropsFromState & ConnectedReduxProps;
+
 export class NavbarBase extends React.Component<Props> {
+  static defaultProps = {
+    _logOutFromServer: logOutFromServer,
+  };
+
+  logOut = async () => {
+    const { _logOutFromServer, apiState, dispatch } = this.props;
+
+    await _logOutFromServer(apiState);
+    dispatch(userActions.logOut());
+  };
+
   render() {
-    const { onLogOut, profile } = this.props;
+    const { profile } = this.props;
```

Comment:
It would be nice if this component had a loading state while it fetches the user profile so I filed https://github.com/mozilla/addons-code-manager/issues/144

Reply:
I submitted #143 for that yesterday (proposal)